syntax = "proto3";

import "google/protobuf/empty.proto";

package helloworld;

// The greeting service definition.
service Greeter {
  // Sends a greeting
  rpc SayHello (HelloRequest) returns (HelloReply) {}

  // Sends a greeting 1 byte a time
  rpc SayHelloSlowly (HelloRequest) returns (stream HelloReply) {}

  rpc Abort(google.protobuf.Empty) returns (google.protobuf.Empty) {}

  rpc UnaryTimeout(TimeoutRequest) returns (google.protobuf.Empty) {}

  rpc StreamTimeout(TimeoutRequest) returns (stream google.protobuf.Empty) {}
}

// The request message containing the user's name.
message HelloRequest {
  string name = 1;
}

// The response message containing the greetings
message HelloReply {
  string message = 1;
}

message TimeoutRequest {
  float seconds = 1;
}
